core/test/variable_test.cpp

```diff
@@ -90,6 +91,23 @@ TEST(Variable, span_references_Variable) {
   EXPECT_EQ(observer[0], 1.0);
 }
 
+using OnDemandType = int16_t; // Not instanitated in cpp.
```

Comment:
I'd use a custom struct defined locally. There is a change that `int16_t` is added to the built-ins later, and then we would need to go back and fix this test.